USA/FL/Bay/Court/scraper/utils/ScraperUtils.py

```diff
@@ -1,57 +1,68 @@
 import os
-import sys
 import csv
 import regex
-from typing import List
-from pathvalidate import sanitize_filename
+from typing import List, Optional
+
+from enforce_typing import enforce_types
+from dataclass_builder import dataclass_builder
 from dataclasses import dataclass
-import requests
-from requests_toolbelt.utils import dump
+from pathvalidate import sanitize_filename
 from requests.exceptions import HTTPError, Timeout
+from requests_toolbelt.utils import dump
+import requests
+from common.pii import Pii
 
 
-@dataclass
+@enforce_types
+@dataclass()
 class Charge:
-    count: int
-    statute: str
-    description: str
-    level: str
-    degree: str
-    disposition: str
-    disposition_date: str
-    offense_date: str
-    citation_number: str
-    plea: str
-    plea_date: str
-
-
-@dataclass
+    count: Optional[int] = None
+    statute: Optional[str] = None
+    description: Optional[str] = None
+    level: Optional[str] = None
+    degree: Optional[str] = None
+    disposition: Optional[str] = None
+    disposition_date: Optional[str] = None
+    offense_date: Optional[str] = None
+    citation_number: Optional[str] = None
+    plea: Optional[str] = None
+    plea_date: Optional[str] = None
+
+
+ChargeBuilder = dataclass_builder(Charge)
+
+
+@enforce_types
+@dataclass(frozen=True)
 class Record:
-    id: str
-    state: str
-    county: str
-    portal_id: str
-    case_num: str
-    agency_report_num: str
-    party_id: str
-    first_name: str
-    middle_name: str
-    last_name: str
-    suffix: str
-    dob: str
-    race: str
-    sex: str
-    arrest_date: str
-    filing_date: str
-    offense_date: str
-    division_name: str
-    case_status: str
-    defense_attorney: str
-    public_defender: str
-    judge: str
-    charges: List[Charge]
-    arresting_officer: str
-    arresting_officer_badge_number: str
+    id: Optional[str]
+    state: Optional[str]
+    county: Optional[str]
```

Comment:
Any practical reasons these fields are Optional, since they need to be used for DB keys?
Surely these should be the minimum required fields?

Reply:
Nope, this was a mistake. Made them mandatory again.